okio/build.gradle

```diff
@@ -1,7 +1,91 @@
-apply plugin: 'org.jetbrains.kotlin.platform.common'
+apply plugin: 'org.jetbrains.kotlin.multiplatform'
 
-dependencies {
-  implementation deps.kotlin.stdLib.common
-  testImplementation deps.kotlin.test.common
-  testImplementation deps.kotlin.test.annotations
+kotlin {
+  targets {
+    // egor: jvmWithJava is getting deprecated but it's not clear what the replacement is. For now 
+    // gonna keep it inside targets {}
+    fromPreset(presets.jvmWithJava, 'jvm') {
+      project.sourceSets.test.java.srcDir('jvm/src/test/java')
+    }
+  }
+  js() {
+    configure([compilations.main, compilations.test]) {
+      tasks.getByName(compileKotlinTaskName).kotlinOptions {
+        moduleKind = "commonjs"
+      }
+    }
+  }
+  iosX64() {
+    compilations.main.outputKinds 'framework'
+  }
+  iosArm64() {
+    compilations.main.outputKinds 'framework'
+  }
+  // Required to generate tests tasks: https://youtrack.jetbrains.com/issue/KT-26547
+  linuxX64()
+  macosX64()
+  mingwX64('winX64')
+  mingwX64()
+  sourceSets {
+    commonMain {
+      kotlin.srcDir('src/main/kotlin')
+      dependencies {
+        implementation deps.kotlin.stdLib.common
+      }
+    }
+    commonTest {
+      kotlin.srcDir('src/test/kotlin')
+      dependencies {
+        implementation deps.kotlin.test.common
+        implementation deps.kotlin.test.annotations
+      }
+    }
+    jvmMain {
+      kotlin.srcDir('jvm/src/main/java')
```

Comment:
Do you also want the resource directory (`jvm/src/main/resources`) for the progruad rules?

Reply:
Yep, good call!